test/specs/testnet/swap/Regression_SingleChainSwap.spec.js

```diff
@@ -538,4 +538,355 @@ describe("The regression suite for the single chain swap on the TestNet", () =>
+  it("Setup the SDK for xDai network and perform the single chain swap action from ERC20 token to ERC20 Token with exceed token balance.", async () => {
+    // Initialize the SDK and define network
+    sdkTestNet = new Sdk(process.env.PRIVATE_KEY, {
+      env: EnvNames.TestNets,
+      networkName: NetworkNames.Xdai,
+    });
+
+    assert.strictEqual(
+      sdkTestNet.state.accountAddress,
+      "0xa5494Ed2eB09F37b4b0526a8e4789565c226C84f",
+      "The EOA Address is not displayed correctly."
+    );
+
+    // Compute the smart wallet address
+    const smartWalletOutput = await sdkTestNet.computeContractAccount();
+    smartWalletAddress = smartWalletOutput.address;
+    console.log("Smart wallet address: ", smartWalletAddress);
+
+    assert.strictEqual(
+      smartWalletAddress,
+      "0x666E17ad27fB620D7519477f3b33d809775d65Fe",
+      "The smart address is not displayed correctly."
+    );
+
+    // GET EXCHANGE OFFERS
+    const offers = await sdkTestNet.getExchangeOffers({
+      fromTokenAddress: "0xDDAfbb505ad214D7b80b1f830fcCc89B60fb7A83", // USDC Token
+      toTokenAddress: "0x4ECaBa5870353805a9F068101A40E0f32ed605C6", // USDT Token
+      // toTokenAddress: ethers.constants.AddressZero,
+      fromAmount: ethers.utils.parseUnits("100000000", 6),
+    });
+
+    // Estimating the batch
+    try {
+      await sdkTestNet.estimateGatewayBatch();
+
+      assert.isFalse(
+        "The Estimation is performed even if exceed the token account."
+      );
+    } catch (e) {
+      console.log(
+        e,
+        "The Estimation is not happened due to exceed token account."
+      );
+    }
+  });
+
+  // SWAP ON XDAI NETWORK FROM ERC20 TOKEN TO NATIVE TOKEN WITH EXCEED TOKEN BALANCE
+  it("Setup the SDK for xDai network and perform the single chain swap action from ERC20 token to native token with exceed token balance.", async () => {
+    // Initialize the SDK and define network
+    sdkTestNet = new Sdk(process.env.PRIVATE_KEY, {
+      env: EnvNames.TestNets,
+      networkName: NetworkNames.Xdai,
+    });
+
+    assert.strictEqual(
+      sdkTestNet.state.accountAddress,
+      "0xa5494Ed2eB09F37b4b0526a8e4789565c226C84f",
+      "The EOA Address is not displayed correctly."
+    );
+
+    // Compute the smart wallet address
+    const smartWalletOutput = await sdkTestNet.computeContractAccount();
+    smartWalletAddress = smartWalletOutput.address;
+    console.log("Smart wallet address: ", smartWalletAddress);
+
+    assert.strictEqual(
+      smartWalletAddress,
+      "0x666E17ad27fB620D7519477f3b33d809775d65Fe",
+      "The smart address is not displayed correctly."
+    );
+
+    // GET EXCHANGE OFFERS
+    const offers = await sdkTestNet.getExchangeOffers({
+      fromTokenAddress: "0xDDAfbb505ad214D7b80b1f830fcCc89B60fb7A83", // USDC Token
+      // fromTokenAddress: ethers.constants.AddressZero,
+      // toTokenAddress: "0x4ECaBa5870353805a9F068101A40E0f32ed605C6", // USDT Token
```

Comment:
Please remember to remove any dead code, or leave a comment into why it's here.

Reply:
Commented code removed from the test cases.